app/controllers/v1/reservations_controller.rb

```diff
@@ -0,0 +1,68 @@
+class V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show edit update destroy]
+
+  # GET /reservations or /reservations.json
+  def index
+    @user = current_user
+    @reservations = Reservation.where(user_id: @user.id)
+    render json: { reservations: @reservations }, status: 200
+  end
+
+  # GET /reservations/1 or /reservations/1.json
+  def show; end
+
+  # GET /reservations/new
+  def new
+    @reservation = Reservation.new
+  end
+
+  # GET /reservations/1/edit
+  def edit
+    @reservation = Reservation.find(params[:id])
+  end
+
+  # POST /reservations or /reservations.json
+  def create
+    @reservation = Reservation.new(
+      user_id: current_user.id,
+      resturant_id: params[:resturant_id],
+      start_time: params[:start_time],
+      end_time: params[:end_time]
+    )
```

Comment:
- [ ] Using your `reservation_params` which is commented below should be even more cleaner than doing this.